Clang compiler unable to link against fortran

### What happened?

Simply executing:

    bazel build --config=clang //geometry:meshcat_manual_test

produces the error message:

```
ERROR: /home/seancurtis/code/drake/geometry/BUILD.bazel:481:16: Linking geometry/meshcat_manual_test failed: (Exit 1): clang failed: error executing command (from target //geometry:meshcat_manual_test) /usr/lib/llvm-12/bin/clang @bazel-out/k8-opt/bin/geometry/meshcat_manual_test-2.params

Use --sandbox_debug to see verbose messages from the sandbox and retain the sandbox build root for debugging
/usr/bin/ld.gold: error: cannot find -lgfortran
clang: error: linker command failed with exit code 1 (use -v to see invocation)
Target //geometry:meshcat_manual_test failed to build
```

However, defaulting to gcc (omitting `--config=clang`) compiles fine. There are no extraneous requests for snopt in any of my local (workspace or user account) bazelrc files.

It has been found that commenting out [this line](https://github.com/RobotLocomotion/drake/pull/15892/files#diff-148d056c20dd1e49668baccad3008ebd63f4e80195df663c6c76a4c9271cd958) alleviates the problem locally (obviously leaving further problems lurking for creating wheels).


### Version

5a5b98c6025994330caf097560da5d3a06b720fe

### What operating system are you using?

Ubuntu 22.04

### What installation option are you using?

compiled from source code using Bazel

### Relevant log output

_No response_

--- a/setup/ubuntu/source_distribution/install_prereqs.sh
+++ b/setup/ubuntu/source_distribution/install_prereqs.sh
@@ -128,9 +128,10 @@ fi
 if [[ "${codename}" == "jammy" ]]; then
   status=$(dpkg-query --show --showformat='${db:Status-Abbrev}' libgcc-12-dev 2>/dev/null || true)
   if [[ "${status}" == "ii " ]]; then
-    status=$(dpkg-query --show --showformat='${db:Status-Abbrev}' libstdc++-12-dev 2>/dev/null || true)
-    if [[ "${status}" != "ii " ]]; then
-      apt-get install ${maybe_yes} --no-install-recommends libgcc-12-dev libstdc++-12-dev
+    status_stdcxx=$(dpkg-query --show --showformat='${db:Status-Abbrev}' libstdc++-12-dev 2>/dev/null || true)
+    status_fortran=$(dpkg-query --show --showformat='${db:Status-Abbrev}' libgfortran-12-dev 2>/dev/null || true)
+    if [[ "${status_stdcxx}" != "ii " || "${status_fortran}" != "ii " ]]; then
+      apt-get install ${maybe_yes} --no-install-recommends libgcc-12-dev libstdc++-12-dev libgfortran-12-dev
     fi
   fi
 fi

